package.json

```diff
@@ -18,19 +18,65 @@
             "type": "object",
             "title": "Doxygen Documentation Generator Settings",
             "properties": {
-                "doxdocgen.generic.commentStart": {
-                    "description": "Doxygen comment start indicator. Default ist /** but Qt style with /*! is also valid.",
+                "doxdocgen.generic.triggerSequence": {
+                    "description": "Doxygen comment trigger. This character sequence triggers generation of DoxyGen comments.",
                     "type": "string",
-                    "default": "/**",
-                    "enum": [
-                        "/**",
-                        "/*!"
-                    ]
+                    "default": "/**"
                 },
-                "doxdocgen.generic.generateReturnType": {
-                    "description": "Insert return type into generated documentation",
+                "doxdocgen.generic.firstLine": {
+                    "description": "The first line of the comment that gets generated. If empty it won't get generated at all.",
+                    "type": "string",
+                    "default": "/**"
+                },
+                "doxdocgen.generic.commentPrefix": {
+                    "description": "The pre fix that is used for each comment line.",
+                    "type": "string",
+                    "default": " * "
+                },
+                "doxdocgen.generic.lastLine": {
+                    "description": "The last line of the comment that gets generated. If empty it won't get generated at all.",
+                    "type": "string",
+                    "default": "**/"
+                },
+                "doxdocgen.generic.newLineAfterBrief": {
+                    "description": "Whether to insert a newline after a brief.",
                     "type": "boolean",
                     "default": true
+                },
+                "doxdocgen.generic.newLineAfterParams": {
+                    "description": "Whether to insert a newline after the params.",
+                    "type": "boolean",
+                    "default": false
+                },
+                "doxdocgen.generic.newLineAfterTParams": {
+                    "description": "Whether to insert a newline after the template params.",
+                    "type": "boolean",
+                    "default": false
+                },
+                "doxdocgen.generic.includeTypeAtReturn": {
+                    "description": "Whether include type information at return.",
+                    "type": "boolean",
+                    "default": false
+                },
+                "doxdocgen.generic.briefTemplate": {
```

Comment:
Generally speaking this templating idea is great I'm just not sure that it's more convenient than setting bool flags if some command should be generated and adding a start parameter option [@, \]?

Reply:
XML style comments are also valid doxygen like this: https://www.stack.nl/~dimitri/doxygen/manual/xmlcmds.html

I don't think we can achieve the desired configuration without adding a templating system.

Reply:
Yes the C# usecase is quite strong.

You're probably right that it might be really tricky to do otherwise.